rich/console.py

```diff
@@ -1102,11 +1106,10 @@ def size(self) -> ConsoleDimensions:
             except OSError:  # Probably not a terminal
                 pass
         else:
-            try:
-                width, height = os.get_terminal_size(sys.__stdin__.fileno())
-            except (AttributeError, ValueError, OSError):
+            for file_descriptor in _STD_STREAMS:
                 try:
-                    width, height = os.get_terminal_size(sys.__stdout__.fileno())
+                    width, height = os.get_terminal_size(file_descriptor)
+                    break
```

Comment:
Suggest putting that break in an `else:` clause.

Reply:
An else: clause on the try except block, avoids having too much code in the try block

```python
try:
    width,height =. ...
except:
    # exceptions
else:
    break
```

Reply:
Ah, I see! Alright, let's do this :+1:  :slightly_smiling_face: